File: app/src/main/java/com/mithril/mobilegoldenleaf/database/dao/ClerkDao.java
package com.mithril.mobilegoldenleaf.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.mithril.mobilegoldenleaf.models.Clerk;

import java.util.List;

@Dao
public interface ClerkDao {
    @Insert
    long save(Clerk address);

    @Update
    int edit(Clerk address);

    @Query("SELECT * FROM Clerk")
    List<Clerk> all();

}

File: app/src/main/java/com/mithril/mobilegoldenleaf/persistence/repository/ClerkDao.kt
package com.mithril.mobilegoldenleaf.persistence.repository

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

import com.mithril.mobilegoldenleaf.models.Clerk

@Dao
interface ClerkDao {
    @Insert
    fun save(address: Clerk): Long

    @Update
    fun edit(address: Clerk): Int

    @Query("SELECT * FROM Clerk")
    fun all(): List<Clerk>

}
